### PR TITLE
Decompose the README & add a contributing guide

### PR DESCRIPTION
**Pull request by Gedochao**
- extracted the dev docs to a separate document
- updated the README header
- linked all (?) side-README docs in the main README, as it was easy to miss some of them
- fixed the formatting here and there
- added a contributing guide
- added some extra explanations on how to filter unit test suites
- updated some other deprecated points (IDE imports now work out of the box, Node 16+ is necessary for the website, etc)

Extra thanks to @alexarchambault, who has written the early draft for the contribution guidelines some time ago.

### CODE REVIEW

File: DEV.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+#### Running the CLI from sources
+
+```bash
+./mill -i scala …arguments…
+```
+
+#### Run unit tests
+
+```bash
+./mill 'build-module.test' './mill 'build-module.test'
```
Comment (lwronski) [inline DEV.md]:
```suggestion
./mill 'build-module.test' 
```

File: DEV.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+```
+
```
Comment (lwronski) [inline DEV.md:32]: Filter unit tests 

 ```bash
./mill 'build-module.test' 'scala.build.tests.DirectiveTests.*'
./mill 'build-module.test' 'scala.build.tests.DirectiveTests.should parse javac options'
 ```

File: DEV.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+Filter test suites with
+
+```bash
+./mill integration.test.jvm 'scala.cli.integration.RunTestsDefault.*'
+./mill integration.test.jvm 'scala.cli.integration.RunTestsDefault.Multiple scripts'
+```
+
+Pass the `--debug` option to debug integration tests:
```
Comment (lwronski) [inline DEV.md]: Describe more how developers can debug Scala CLI, when pass `--debug` option to integrate test.

File: DEV.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+```bash
+./mill integration.test.jvm 'scala.cli.integration.RunTestsDefault.*' --debug:5006
+```
+
+#### Run integration tests with the native launcher
+
+(generating the launcher can take several minutes)
+
+```bash
+./mill integration.test.native
```
Comment (lwronski) [inline DEV.md:64]:
```suggestion
./mill integration.test.native
./mill integration.test.native 'scala.cli.integration.RunTestsDefault.*'
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+## Pull requests
+
+### Fork-Pull
+
+We accept external pull requests according to
+the [standard GitHub fork-pull flow](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
+Create a fork of this repository, commit your changes there and create a pull request. We try to review those as often
+as possible.
+
+### Rules for a well-formed PR
```
Comment (lwronski) [inline CONTRIBUTING.md:33]: Add rules when someone create PR with changes in documentation. In this case developers should open PR to `stable` branch.

File: DEV.md (hunk 2 above)
Comment (Gedochao) [reply DEV.md:32]: was added in a separate commit (1ad4188bda262b5864d109c67c83bbe596a938ff), but I can merge the filter examples into a single code block...

File: DEV.md (hunk 3 above)
Comment (Gedochao) [reply DEV.md]: should be fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (Gedochao) [reply CONTRIBUTING.md:33]: added a separate section on the `main` and `stable` branches, please verify if that's sufficient

**Review by lwronski — APPROVED** [review]
LGTM

**Review by tgodzik — APPROVED** [review]

**Pull request merged**